### PR TITLE
log-backup: add timeout for operations that may stuck

### LINKED ISSUE #13251: PiTR log backup checkpoint doesn't move foward
### What version of TiKV are you using?
[root@localhost bin]# ./tikv-server -V
TiKV
Release Version:   6.2.0-alpha
Edition:           Community
Git Commit Hash:   8a4c8b3828fe689e977c671796381d2e9b16bc07
Git Commit Branch: heads/refs/tags/v6.2.0
UTC Build Time:    2022-08-02 12:18:20
Rust Version:      rustc 1.62.0-nightly (7c4b47696 2022-04-30)
Enable Features:   pprof-fp jemalloc mem-profiling portable sse test-engine-kv-rocksdb test-engine-raft-raft-engine cloud-aws cloud-gcp cloud-azure
Profile:           dist_release

### What operating system and CPU are you using?
Cent OS, 8C 64GB

### Steps to reproduce
1. There is a br log backup task running for 15 days.

### What did you expect?
Checkpoint should move forward correctly

### What did happened?
Checkpoint not move forward somehow.

image

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #13251

What's Changed:

This PR added timeout for some operations so we won't get stuck if there are some network issues.

Because we cannot collect enough information when log backup get stuck, we add timeout for two places:
- Every RPC call to etcd.
- The handler `on_update_global_checkpoint`. <del>(This also moved it to background.)</del>

This PR also set the `keep_alive_while_idle` for tonic. So PD won't cut down the connection with error `too many pings`.

For adding timeout for the RPCs, we need to cherry-pick some commits from newer version of the etcd client, and for http2 keep alive while idle config, we need a tiny patch, you can check the diff [here](https://github.com/etcdv3/etcd-client/compare/0.7.2...YuJuncen:etcd-client:v0.7.2-patch-extra-options?expand=1).

```commit-message
Added a timeout of `30s` for every etcd gRPC request.
Moved `on_update_global_checkpoint` to background.
```

### Check List 

Tests 

- Manual test (add detailed scripts or steps below)

### Release note 

```release-note
None
```

### CODE REVIEW

File: components/backup-stream/Cargo.toml [hunk 1]
```diff
@@ -29,7 +29,9 @@ dashmap = "5"
 engine_rocks = { path = "../engine_rocks", default-features = false }
 engine_traits = { path = "../engine_traits", default-features = false }
 error_code = { path = "../error_code" }
-etcd-client = { version = "0.7", features = ["pub-response-field", "tls"] }
+# We cannot update the etcd-client to latest version because of the cyclic requirement.
```
Comment (3pointer) [inline components/backup-stream/Cargo.toml:32]: maybe leave a TODO here, once we can update etcd-client.

**Review by 3pointer — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Comment by joccau** [comment]
Update PR information, this pr has integration test and manual test in TiDB cluster with log-backup task.

**Pull request merged**